### PR TITLE
Implement best tip block hashes for ChainTip receivers

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

In ticket #2639, we want to get block hashes, transaction information, and rollback information out of the state.

The `ChainTip` struct already gets block heights out of the state.

In this PR, I've added block hashes to the `ChainTip` information.

## Solution

- Always prefer the non-finalized tip in ChainTipSender
- Update ChainTipSender with blocks, not heights
- Provide a best tip hash in ChainTip receivers

Optimisation:
- Skip finalized blocks once the non-finalized state is active 

This PR does not close any tickets.

## Review

@jvff can review this PR.

This PR is based on PR #2676.

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

## Follow Up Work

Do the rest of ticket #2639: transaction IDs, resets, initialization and missed update handling.

### CODE REVIEW

**Comment by teor2345** [comment]
@jvff this is ready for review now, I'll do the optimisations in a separate PR.

**Review by jvff — DISMISSED** [review]

File: zebra-state/src/service/chain_tip/tests/prop.rs [hunk 1]
```diff
@@ -1,47 +1,68 @@
...
 
+const DEFAULT_BLOCK_VEC_PROPTEST_CASES: u32 = 4;
+
 proptest! {
+    #![proptest_config(
+        proptest::test_runner::Config::with_cases(env::var("PROPTEST_CASES")
+            .ok()
+            .and_then(|v| v.parse().ok())
+            .unwrap_or(DEFAULT_BLOCK_VEC_PROPTEST_CASES))
+    )]
```
Comment (dconnolly) [inline zebra-state/src/service/chain_tip/tests/prop.rs:18]: 👍

File: zebra-state/src/service/finalized_state.rs [hunk 2]
```diff
@@ -473,6 +509,12 @@ impl FinalizedState {
             })
     }
 
+    /// Returns the tip block, if there is one.
+    pub fn tip_block(&self) -> Option<Arc<Block>> {
+        let (height, _hash) = self.tip()?;
+        self.block(height.into())
+    }
```
Comment (dconnolly) [inline zebra-state/src/service/finalized_state.rs:516]: 👍

File: zebra-state/src/service/non_finalized_state.rs [hunk 3]
```diff
@@ -318,6 +318,12 @@ impl NonFinalizedState {
         Some((height, hash))
     }
 
+    /// Returns the block at the tip of the best chain.
+    pub fn best_tip_block(&self) -> Option<Arc<Block>> {
+        let (height, _hash) = self.best_tip()?;
+        self.best_block(height.into())
+    }
```
Comment (dconnolly) [inline zebra-state/src/service/non_finalized_state.rs:325]: 👍

**Review by dconnolly — APPROVED** [review]

**Pull request merged**